code/modules/disposals/disposal_test.dm

```diff
@@ -0,0 +1,91 @@
+// How 2 use: run /proc/test_disposal_system via Advanced ProcCall
+// locate the X and Y where normally disposed stuff should end up (usually the last bit of conveyor belt in front of the crusher door
+// wait and see what comes up!
+/proc/test_disposal_system(var/expected_x, var/expected_y, var/sleep_time = 1200, var/include_mail = TRUE)
+	if (!usr && (isnull(expected_x) || isnull(expected_y)))
+		return
+	if (isnull(expected_x))
+		expected_x = input(usr,"Please enter X coordinate") as null|num
+		if (isnull(expected_x))
+			return
+	if (isnull(expected_y))
+		expected_y = input(usr,"Please enter Y coordinate") as null|num
+		if (isnull(expected_y))
+			return
+
+	var/list/dummy_list = list()
+	for_by_tcl(D, /obj/machinery/disposal)
+		if (!inonstationz(D))
+			break
+		/*
+		if (D.type != text2path(test_path))
+			continue
+		*/
+		var/obj/item/disposal_test_dummy/TD
+		// Mail chute test
+		if(istype(D, /obj/machinery/disposal/mail))
+			if(include_mail)
+				var/obj/machinery/disposal/mail/mail_chute = D
+
+				SPAWN(2 SECONDS)
+					for(var/dest in mail_chute.destinations)
+						var/obj/item/disposal_test_dummy/mail_test/MD = new /obj/item/disposal_test_dummy/mail_test(mail_chute, sleep_time)
+						MD.source_disposal = mail_chute
+						MD.destination_tag = dest
+						mail_chute.destination_tag = dest
+						dummy_list.Add(MD)
+						mail_chute.flush()
```

Comment:
I'm confused why this sleeps for 2 seconds? Seems like it'll just wait 2 seconds and then rapid fire spawn/flush all the items right after each other, so it's not reducing lag or anything.

Reply:
without it the process gets all fucked up and broken cuz it waits for all that stuff to be done before continuining, it could be a SPAWN(0) but it was 2 seconds in the original code